src/Compilers/CSharp/Test/Symbol/Symbols/FunctionPointerTypeSymbolTests.cs

```diff
@@ -758,5 +647,48 @@ class C
                     //     void M(delegate*<in string, void> p1) {}
                     Diagnostic(ErrorCode.ERR_PredefinedTypeNotFound, "in string").WithArguments("System.Runtime.InteropServices.InAttribute").WithLocation(4, 22));
         }
+
+        [Fact]
+        public void DifferringModOpts()
+        {
+            var ilSource = @"
+.class public auto ansi beforefieldinit Test1
+       extends[mscorlib] System.Object
+{
+    .method public hidebysig instance void  M(method bool& *(int32&) noModopts,
+                                              method bool modopt([mscorlib]System.Object)& *(int32&) modoptOnReturn,
+                                              method bool& *(int32 modopt([mscorlib]System.Object)&) modoptOnParam,
+                                              method bool& modopt([mscorlib]System.Object) *(int32&) modoptOnReturnRef,
+                                              method bool& *(int32& modopt([mscorlib]System.Object)) modoptOnParamRef) cil managed
+    {
+      // Code size       2 (0x2)
+      .maxstack  8
+      IL_0000:  nop
+      IL_0001:  ret
+    } // end of method Test::M
+}
+";
+
+            var comp = CreateCompilationWithIL("", ilSource, parseOptions: TestOptions.RegularPreview);
+            var testClass = comp.GetTypeByMetadataName("Test1")!;
+            var m = testClass.GetMethod("M");
+
+            foreach (var param1 in m.Parameters)
+            {
+                foreach (var param2 in m.Parameters)
+                {
+                    if (!ReferenceEquals(param1, param2))
+                    {
+                        Assert.False(param1.Type.Equals(param2.Type, TypeCompareKind.ConsiderEverything));
+                    }
+                    else
+                    {
+                        Assert.True(param1.Type.Equals(param2.Type, TypeCompareKind.ConsiderEverything));
+                    }
```

Comment:
```
Assert.NotEqual(
    ReferenceEquals(param1, param2),
    param1.Type.Equals(param2.Type, TypeCompareKind.ConsiderEverything));
```

Reply:
I think the way it's written currently is easier to understand and diagnose if something went wrong. I'm going to leave it as is.

---
In reply to: [374345105](https://github.com/dotnet/roslyn/pull/41340#discussion_r374345105) [](ancestors = 374345105)